chat.go

```diff
@@ -55,7 +56,26 @@ type ChatCompletionRequest struct {
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
 	Functions        []*FunctionDefine       `json:"functions,omitempty"`
-	FunctionCall     string                  `json:"function_call,omitempty"`
+	// FunctionCall can either be a string ("none" or "auto") or an instance of ChatCompletionRequestFunctionCall
+	FunctionCall any `json:"function_call,omitempty"`
+}
+
+func checkFunctionCall(value any) bool {
```

Comment:
Hey, thank you for the PR! The name of this function sounds a bit too vague. Maybe let's go with something a bit more specific like `checkIsChatCompletionFunctionCall`?

Reply:
@sashabaranov Good idea - I've just pushed a change to rename the function to exactly that.